azurerm/data_source_cosmos_db_account.go

```diff
@@ -0,0 +1,218 @@
+						},
+					},
+				},
+			},
+
+			"geo_location": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"location": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"failover_priority": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"read_endpoints": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"write_endpoints": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"primary_master_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"secondary_master_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"primary_readonly_master_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"secondary_readonly_master_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceArmCosmosDBAccountRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).cosmosDBClient
+	ctx := meta.(*ArmClient).StopContext
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on AzureRM CosmosDB Account '%s': %s", name, err)
+	}
+
+	d.SetId(*resp.ID)
+	d.Set("name", resp.Name)
+	d.Set("location", azureRMNormalizeLocation(*resp.Location))
+	d.Set("resource_group_name", resourceGroup)
+	flattenAndSetTags(d, resp.Tags)
+
+	d.Set("kind", string(resp.Kind))
+	d.Set("offer_type", string(resp.DatabaseAccountOfferType))
+	d.Set("ip_range_filter", resp.IPRangeFilter)
+	d.Set("endpoint", resp.DocumentEndpoint)
+
+	if v := resp.EnableAutomaticFailover; v != nil {
+		d.Set("enable_automatic_failover", resp.EnableAutomaticFailover)
+	}
+
+	if err := d.Set("consistency_policy", flattenAzureRmCosmosDBAccountConsistencyPolicy(resp.ConsistencyPolicy)); err != nil {
+		return fmt.Errorf("Error setting CosmosDB Account %q `consistency_policy` (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	//sort `geo_locations` by fail over priority
+	locations := make([]map[string]interface{}, len(*resp.FailoverPolicies))
+	for _, l := range *resp.FailoverPolicies {
+		locations[*l.FailoverPriority] = map[string]interface{}{
+			"id":                *l.ID,
+			"location":          azureRMNormalizeLocation(*l.LocationName),
+			"failover_priority": int(*l.FailoverPriority),
+		}
+	}
+	d.Set("geo_location", &locations)
```

Comment:
we shouldn't need the `&` here since `d.Set` will take care of the referencing for us